src/logic/transactionEngine.ts

```diff
@@ -22,32 +54,41 @@ export async function createSplTransfer(sender: string, destination: string, amo
     mint.address
   );
 
+  const amountLamports = mint.decimals === 0 ? amount : new Decimal(amount).mul(new Decimal(10).pow(mint.decimals)).toString();
```

Comment:
note: you can update ix_TransferSPL to accept the int decimals. you should probably create a transfer checked instruction anyway

Reply:
Agree. instruction creation should probably not be part of the core library.
Should be either:

- Separate library from `tinywallet`
- Just another Typescript module in this project.

I'll do the latter on a separate PR.